iomgr/lib/dbengine/handlers/RequestHandler_DQL.cpp

```diff
@@ -506,5 +506,71 @@ void RequestHandler::executeShowTablesRequest(iomgr_protocol::DatabaseEngineResp
     codedOutput.WriteVarint64(kNoMoreRows);
     rawOutput.CheckNoError();
 }
+void RequestHandler::executeDescribeTableRequest(iomgr_protocol::DatabaseEngineResponse& response,
+        [[maybe_unused]] const requests::DescribeTableRequest& request)
+{
+    response.set_has_affected_row_count(false);
+    response.set_affected_row_count(0);
+
+    const auto& databaseName =
+            request.m_database.empty() ? m_currentDatabaseName : request.m_database;
+    const auto database = m_instance.findDatabaseChecked(databaseName);
+    UseDatabaseGuard databaseGuard(*database);
+    const auto table = database->findTableChecked(request.m_table);
+
+    const auto sysColumnsTableName = database->findTableChecked(kSysColumnsTableName);
+    const auto nameColumn = sysColumnsTableName->findColumnChecked(kSysColumns_Name_ColumnName);
+    const auto dataTypeColumn =
+            sysColumnsTableName->findColumnChecked(kSysColumns_DataType_ColumnName);
+
+    addColumnToResponse(response, *nameColumn, "");
+    auto columnDescription = response.add_column_description();
+    columnDescription->set_name("DATA_TYPE");
+    columnDescription->set_is_null(false);
+    columnDescription->set_type(ColumnDataType::COLUMN_DATA_TYPE_TEXT);
+
+    const bool nullNotAllowed = nameColumn->isNotNull() && dataTypeColumn->isNotNull();
+
+    const auto columnRecords = database->getColumnsRecordsOrderedByName(table->getId());
```

Comment:
The correct logic is to do ```const auto columns = table->getColumnsOrderedByPosition();``` and then iterate over it: `for (const auto& column: columns)`. Get column name as `column->getName()`, get column data type as `column->getDataType()` (it returns numeric type id, so you still need to call `getColumnDataTypeName()`).